### PR TITLE
LATAM HOC promo video on hourofcode.com homepage

### PR DESCRIPTION
**Pull request by Erin007**
Users viewing the site in Spanish(Mexico) "la", Spanish(Spain) "es", or Portuguese(Brazil) "pt", will see a different promotional video on the Hour of Code homepage to promote Latin American Hour of Code. Users in all other languages are not affected.

Similar to #25023

### CODE REVIEW

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 1]
```diff
@@ -232,3 +245,31 @@ social:
       caption.fadeOut();
     });
   });
+
+  function startLATAMVideo()
+  {
+    if (videoStarted)
+      return;
+
+    $("<iframe allowfullscreen frameborder='0' src='https://www.youtube-nocookie.com/embed/OpgWbXthMlQ?iv_load_policy=3&amp;rel=0&amp;autohide=1&amp;showinfo=0&amp;autoplay=1&cc_load_policy=1' style='position:absolute; top: 0; left: 0; width: 100%; height: 100%; background-color:black;'></iframe>").appendTo("#videodiv");
```
Comment (Erin007) [inline pegasus/sites.v3/hourofcode.com/public/index.haml]: I'm still waiting on the video code - right now this plays a Finding Nemo clip 🐠. Double check that I update when the video is ready!

**Comment by Erin007** [comment]
I did this in a way that duplicates a big chunk of code for a few reasons: 1.) It was fastest 2.) I knew it wouldn't break the current video for other languages 3.) It's temporary, so keeping everything separate rather than parameterized makes it easier to delete later.  4.) It was fastest. 

I'm open to other suggestions if we think the duplication is too gross to live with.

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 2]
```diff
@@ -232,3 +245,31 @@ social:
...
+    $("#leftbutton").fadeTo(500, 0).remove();
+
+    $("#fullwidth" + currentImage).fadeTo(1000, 0.3);
+
+    videoStarted = true;
+
+    clearInterval(timerId);
+  }
+  $(function() {
+    $('.highlight-item').mouseover(function() {
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/index.haml:270]: Do we expect the caption to be available on touch devices in some way?

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 3]
```diff
@@ -87,8 +87,14 @@ social:
                 = view :index_video
                 %div{style: "clear: both;"}
               #leftbutton.visible-xs
-                %h1.front-header-banner.watch-video=hoc_s(:front_watch_regular_video)
-                %img{src: "/images/play-button.png", width: "60", onclick: "return startVideo();", style: "cursor: pointer;"}
+                -# If language is Espanol(Mexico), show alternate promo video
+                -# for LATM Hour of Code.
+                - unless @language == "la"
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/index.haml]: Deliberately not doing an `else` to complement the new `if`?

**Comment by breville** [comment]
This seems fine if it's truly temporary code.  But if there's a chance that this kind of change becomes an annual occurrence we should think about how to drive it by DCDO or some other kind of configuration rather than committing code swaps repeatedly, of course.

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 4]
```diff
@@ -101,9 +107,20 @@ social:
...
+              -# for LATM Hour of Code.
+              - unless @language == "la"
+                %div.watch-video-small{onclick: "return startVideo();"}
+                  =hoc_s(:front_watch_regular_video)
+                  %img{src: "/images/play-button.png", width: "25", onclick: "return startVideo();", style: "cursor: pointer;"}
+              - if @language == "la"
+                %div.watch-video-small{onclick: "return startLATAMVideo();"}
+                  =hoc_s(:front_watch_regular_video)
+                  %img{src: "/images/play-button.png", width: "25", onclick: "return startLATAMVideo();", style: "cursor: pointer;"}
+              - else
```
Comment (Hamms) [inline pegasus/sites.v3/hourofcode.com/public/index.haml:120]: using all three of `unless`, `if`, and `else` seems redundant, unless I'm misunderstanding something; won't this render the video twice for non-LA?

Comment (Erin007) [reply pegasus/sites.v3/hourofcode.com/public/index.haml:120]: Done!

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 5]
```diff
@@ -14,6 +14,10 @@ social:
 -# whenever we change the hoc_mode, to make sure we're still testing what we'll see on production.
 -hoc_mode = DCDO.get("hoc_mode", CDO.default_hoc_mode)
 
+-# Spanish(Mexico), Spanish(Spain), and Portuguese(Brazil)
```
Comment (Hamms) [inline pegasus/sites.v3/hourofcode.com/public/index.haml:17]: Is it intentional that we're showing Latin American content for a European locale?

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 6]
```diff
@@ -87,8 +91,14 @@ social:
                 = view :index_video
                 %div{style: "clear: both;"}
               #leftbutton.visible-xs
-                %h1.front-header-banner.watch-video=hoc_s(:front_watch_regular_video)
-                %img{src: "/images/play-button.png", width: "60", onclick: "return startVideo();", style: "cursor: pointer;"}
+                -# If language Spanish or Brazilian Portuguese, show alternate
+                -# promo video for LATM Hour of Code.
+                - if latam_language_codes.include?(@language) && show_latam_videos
+                  %h1.front-header-banner.watch-video=hoc_s(:front_watch_regular_video)
+                  %img{src: "/images/play-button.png", width: "60", onclick: "return startLATMVideo();", style: "cursor: pointer;"}
```
Comment (tanyaparker) [inline pegasus/sites.v3/hourofcode.com/public/index.haml]: Typo `startLATMVideo`

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 5 above)
Comment (Erin007) [reply pegasus/sites.v3/hourofcode.com/public/index.haml:17]: Yes, per a request from Suky/Hadi we want all Spanish locales to see this video.

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 6 above)
Comment (Erin007) [reply pegasus/sites.v3/hourofcode.com/public/index.haml]: Ah! Good catch - I did that with the branch names too 🙃

File: pegasus/sites.v3/hourofcode.com/public/index.haml [hunk 7]
```diff
@@ -87,8 +91,14 @@ social:
                 = view :index_video
                 %div{style: "clear: both;"}
               #leftbutton.visible-xs
-                %h1.front-header-banner.watch-video=hoc_s(:front_watch_regular_video)
-                %img{src: "/images/play-button.png", width: "60", onclick: "return startVideo();", style: "cursor: pointer;"}
+                -# If language Spanish or Brazilian Portuguese, show alternate
+                -# promo video for LATM Hour of Code.
```
Comment (tanyaparker) [inline pegasus/sites.v3/hourofcode.com/public/index.haml]: Typo: `LATM`

**Comment by tanyaparker** [comment]
I don't know if I would say that this is temporary. Even if we take it down for a bit while we show some other videos for other hoc things we're announcing this year, we'll want to have this video be the LATAM video for the rest of the year.

File: pegasus/sites.v3/hourofcode.com/public/index.haml (hunk 2 above)
Comment (tanyaparker) [reply pegasus/sites.v3/hourofcode.com/public/index.haml:270]: How come this part has to be repeated even though it's exactly the same as the original video and it's outside of the startLATAMVideo() function? (I may just not understand Javascript)

```
$(function() {
    $('.highlight-item').mouseover(function() {
      var caption = $(this).find('.highlight-caption');
      caption.fadeIn();
    });

    $('.highlight-item').mouseleave(function() {
      var caption = $(this).find('.highlight-caption');
      caption.fadeOut();
    });
  });
```

**Comment by tanyaparker** [comment]
Can you explain why we already have two places to start the video? What's the difference between the `#leftbutton` and `#remainder` sections?

**Comment by Erin007** [comment]
I think the different places to start the video are for the mobile and desktop versions of the page?

On small screens, the play button is scrunched into a `#leftbutton` div 
screen shot 2018-10-01 at 12 30 44 pm

On desktop it's in  `watch-video-small` container
screen shot 2018-10-01 at 12 31 55 pm

**Comment by Erin007** [comment]
I'm going to get this in as is today, and then re-visit for a more long term refactor to deduplicate and more gracefully handle hiding/showing these videos.

**Comment by tanyaparker** [comment]
Oh, gotcha. Thanks for explaining!

**Pull request merged**